### PR TITLE
gh-108374: Add --disable-gil option to PCbuild/build.bat

### PR DESCRIPTION
**Pull request by colesbury**
This adds a `--disable-gil` option to PCbuild/build.bat. For now, all this does is define the `Py_NOGIL` macro.

* Issue: gh-108374

### CODE REVIEW

**Comment by colesbury** [comment]
I'm not familiar with msbuild, so I'd appreciate feedback if this is the correct approach.

**Comment by zware** [comment]
This looks reasonable to me, but should definitely get signoff from @zooba before merge.

**Review by zooba — APPROVED** [review]
LGTM

**Pull request merged**
